### ISSUE TITLE
multitasker breaks depth and rdepth

### ISSUE THREAD

**Issue by tve** [jeelabs/mecrisp-stellaris#3]
Example:
```
Mecrisp-Stellaris RA 2.3.5 with M0 core for STM32L053C8 by Matthias Koch
64 KB <jz1-tve> 39440C4E ram/flash: 5700 16176 free ok.
task: test-task  ok.
: test-loop test-task activate  begin rdepth . depth . cr  1000 ms again ;  ok.
multitask test-loop  10o7k3.74
1636 1073741635
1073741636 1073741635
1073741636 1073741635
1073741636 1073741635
```
(You can also see the garbled output problem in the stuff printed after `test-loop`. That was supposed to read `multitask test-loop ok`

**Comment by Mecrisp** [comment]
Redefining depth and rdepth is necessary for use in other tasks. You'll also need new .s  u.s  h.s utilities.


\ Multitasking debug tools

:  depth ( -- n ) up @ boot-task = if  depth    else up @ 4 cells stackspace    + + sp@ - 2 arshift then ;
: rdepth ( -- n ) up @ boot-task = if rdepth 1- else up @ 4 cells stackspace 2* + + rp@ - 2 arshift then ;

: .s ( -- )
  base @ >r decimal depth ." Stack: [" . ." ] " r> base !
  depth >r
  begin
    r@ 0 >
  while
    r@ pick .
    r> 1- >r
  repeat
  rdrop
  ."  TOS: " dup . ."  *>" cr
;

: u.s ( -- )
  base @ >r decimal depth ." Stack: [" . ." ] " r> base !
  depth >r
  begin
    r@ 0 >
  while
    r@ pick u.
    r> 1- >r
  repeat
  rdrop
  ."  TOS: " dup u. ."  *>" cr
;

: h.s ( -- )
  base @ >r decimal depth ." Stack: [" . ." ] " r> base !
  depth >r
  begin
    r@ 0 >
  while
    r@ pick hex.
    r> 1- >r
  repeat
  rdrop
  ."  TOS: " dup hex. ."  *>" cr
;

**Comment by jcw** [comment]
Solved by adding the above code to 2.3.7, I think - closing.
